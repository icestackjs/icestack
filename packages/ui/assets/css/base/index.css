:root {
  --primary-content: 255 255 255;
  --primary: 25 137 250;
  --success-content: 255 255 255;
  --success: 7 193 96;
  --error-content: 255 255 255;
  --error: 238 10 36;
  --warning-content: 255 255 255;
  --warning: 255 151 106;
}