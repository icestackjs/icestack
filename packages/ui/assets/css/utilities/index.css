.mp-reset-button {
  padding: 0;
  background-color: transparent;
  /* #ifndef APP-PLUS */
  font-size: inherit;
  line-height: inherit;
  color: inherit;
  /* #endif */
  /* #ifdef APP-NVUE */
  border-width: 0;
  /* #endif */
}

/* #ifndef APP-NVUE */
.mp-reset-button::after {
  border: none;
}

/* #endif */