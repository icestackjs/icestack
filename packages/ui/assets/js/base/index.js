module.exports = {
  ":root": {
    "--primary-color": "255 255 255",
    "--primary": "25 137 250",
    "--success-color": "255 255 255",
    "--success": "7 193 96",
    "--error-color": "255 255 255",
    "--error": "238 10 36",
    "--warning-color": "255 255 255",
    "--warning": "255 151 106"
  }
}