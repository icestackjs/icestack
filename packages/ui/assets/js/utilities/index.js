module.exports = {
  ".mp-reset-button": {
    "padding": "0",
    "backgroundColor": "transparent",
    "fontSize": "inherit",
    "lineHeight": "inherit",
    "color": "inherit",
    "borderWidth": "0"
  },
  ".mp-reset-button::after": {
    "border": "none"
  }
}