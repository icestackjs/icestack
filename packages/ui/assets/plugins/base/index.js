const _plugin = require("tailwindcss/plugin");
const returnSelfNoop = x => x;
const css2TwPlugin = _plugin.withOptions(function (_options = {}) {
  const {
    withOptionsWalkCSSRuleObject = returnSelfNoop
  } = _options;
  return function ({
    addBase: addBase,
    addComponents: addComponents,
    addUtilities: addUtilities,
    theme: theme,
    addVariant: addVariant,
    config: config,
    corePlugins: corePlugins,
    e: e,
    matchComponents: matchComponents,
    matchUtilities: matchUtilities,
    matchVariant: matchVariant
  }) {
    const _baseCss = withOptionsWalkCSSRuleObject({
      ":root": {
        "--primary-content": "255 255 255",
        "--primary": "25 137 250",
        "--success-content": "255 255 255",
        "--success": "7 193 96",
        "--error-content": "255 255 255",
        "--error": "238 10 36",
        "--warning-content": "255 255 255",
        "--warning": "255 151 106"
      }
    }, "base");
    addBase(_baseCss);
    const _componentsCss = withOptionsWalkCSSRuleObject({}, "components");
    addComponents(_componentsCss);
    const _utilitiesCss = withOptionsWalkCSSRuleObject({}, "utilities");
    addUtilities(_utilitiesCss);
  };
}, function (_options) {
  return {};
});
module.exports = css2TwPlugin;