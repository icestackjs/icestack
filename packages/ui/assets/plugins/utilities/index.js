const _plugin = require("tailwindcss/plugin");
const returnSelfNoop = x => x;
const css2TwPlugin = _plugin.withOptions(function (_options = {}) {
  const {
    withOptionsWalkCSSRuleObject = returnSelfNoop
  } = _options;
  return function ({
    addBase: addBase,
    addComponents: addComponents,
    addUtilities: addUtilities,
    theme: theme,
    addVariant: addVariant,
    config: config,
    corePlugins: corePlugins,
    e: e,
    matchComponents: matchComponents,
    matchUtilities: matchUtilities,
    matchVariant: matchVariant
  }) {
    const _baseCss = withOptionsWalkCSSRuleObject({}, "base");
    addBase(_baseCss);
    const _componentsCss = withOptionsWalkCSSRuleObject({}, "components");
    addComponents(_componentsCss);
    const _utilitiesCss = withOptionsWalkCSSRuleObject({
      ".mp-reset-button": {
        "padding": "0",
        "background-color": "transparent",
        "font-size": "inherit",
        "line-height": "inherit",
        "color": "inherit",
        "border-width": "0"
      },
      ".mp-reset-button::after": {
        "border": "none"
      }
    }, "utilities");
    addUtilities(_utilitiesCss);
  };
}, function (_options) {
  return {};
});
module.exports = css2TwPlugin;