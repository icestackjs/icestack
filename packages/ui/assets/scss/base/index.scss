@use "../utils/index.scss" as *;
@use "sass:map";

@each $theme, $map in injectModeSelectors() {
  #{map.get($map,"selector")} {
    @each $var, $value in injectCssVars($theme) {
      #{$var}: $value;
    }
  }
}
