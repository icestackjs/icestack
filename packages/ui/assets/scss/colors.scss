$root-vars: (
  --primary-content: #ffffff,
  --primary: #1989fa,
  --success-content: #ffffff,
  --success: #07c160,
  --error-content: #ffffff,
  --error: #ee0a24,
  --warning-content: #ffffff,
  --warning: #ff976a
);

