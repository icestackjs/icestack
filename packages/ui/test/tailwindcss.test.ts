import plugin from 'tailwindcss/plugin'
import merge from 'merge'
import base from '../assets/js/base/index.js'
import components from '../assets/js/components/index.js'
import utilities from '../assets/js/utilities/index.js'
import { colors } from '@/colors.js'
import { groupBy } from '@/utils'

describe('tailwindcss', () => {
  it('grouped components', () => {
    const dic = groupBy(Object.entries(components), ([name]) => {
      let com = name
      if (com.includes('/')) {
        com = name.split('/')[1]
      }
      return com
    })
    for (const key in dic) {
      if (Object.prototype.hasOwnProperty.call(dic, key)) {
        const arr = dic[key]
        // @ts-ignore
        dic[key] = arr.map((x) => x[1])
      }
    }
    expect(dic).toBeTruthy()
  })

  it('entries components', () => {
    const entries = Object.entries(
      groupBy(Object.entries(components), ([name]) => {
        let com = name
        if (com.includes('/')) {
          com = name.split('/')[1]
        }
        return com
      })
    ).map(([key, arr]) => {
      return [key, arr.map((x) => x[1])]
    })
    for (const [name, c] of entries) {
      if (Array.isArray(c)) {
        expect(merge.recursive(true, ...c)).toMatchSnapshot('merged')
        for (const cc of c) {
          if (typeof cc === 'string') {
            console.log(name, cc)
          }
        }
      }
    }

    expect(entries).toBeTruthy()
    // expect(entries).toMatchSnapshot('common')
  })

  it('utilitiesEntries', () => {
    const utilitiesEntries = Object.entries(
      groupBy(Object.entries(utilities), ([name]) => {
        let com = name
        if (com.includes('/')) {
          com = name.split('/')[1]
        }
        return com
      })
    ).map(([key, arr]) => {
      return [key, arr.map((x) => x[1])]
    })

    for (const [name, u] of utilitiesEntries) {
      if (Array.isArray(u)) {
        expect(merge.recursive(true, ...u)).toMatchSnapshot()
      }
    }
  })
})
