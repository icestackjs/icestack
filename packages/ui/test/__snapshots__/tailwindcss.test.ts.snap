// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tailwindcss > use plugin case 0 1`] = `
"*, ::before, ::after {--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x:  ;--tw-pan-y:  ;--tw-pinch-zoom:  ;--tw-scroll-snap-strictness: proximity;--tw-gradient-from-position:  ;--tw-gradient-via-position:  ;--tw-gradient-to-position:  ;--tw-ordinal:  ;--tw-slashed-zero:  ;--tw-numeric-figure:  ;--tw-numeric-spacing:  ;--tw-numeric-fraction:  ;--tw-ring-inset:  ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / 0.5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur:  ;--tw-brightness:  ;--tw-contrast:  ;--tw-grayscale:  ;--tw-hue-rotate:  ;--tw-invert:  ;--tw-saturate:  ;--tw-sepia:  ;--tw-drop-shadow:  ;--tw-backdrop-blur:  ;--tw-backdrop-brightness:  ;--tw-backdrop-contrast:  ;--tw-backdrop-grayscale:  ;--tw-backdrop-hue-rotate:  ;--tw-backdrop-invert:  ;--tw-backdrop-opacity:  ;--tw-backdrop-saturate:  ;--tw-backdrop-sepia:  
}::backdrop {--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x:  ;--tw-pan-y:  ;--tw-pinch-zoom:  ;--tw-scroll-snap-strictness: proximity;--tw-gradient-from-position:  ;--tw-gradient-via-position:  ;--tw-gradient-to-position:  ;--tw-ordinal:  ;--tw-slashed-zero:  ;--tw-numeric-figure:  ;--tw-numeric-spacing:  ;--tw-numeric-fraction:  ;--tw-ring-inset:  ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / 0.5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur:  ;--tw-brightness:  ;--tw-contrast:  ;--tw-grayscale:  ;--tw-hue-rotate:  ;--tw-invert:  ;--tw-saturate:  ;--tw-sepia:  ;--tw-drop-shadow:  ;--tw-backdrop-blur:  ;--tw-backdrop-brightness:  ;--tw-backdrop-contrast:  ;--tw-backdrop-grayscale:  ;--tw-backdrop-hue-rotate:  ;--tw-backdrop-invert:  ;--tw-backdrop-opacity:  ;--tw-backdrop-saturate:  ;--tw-backdrop-sepia:  
}.btn {display: inline-flex;flex-shrink: 0;cursor: pointer;user-select: none;flex-wrap: wrap;align-items: center;justify-content: center;border-radius: var(--ice-rounded-btn, 0.5rem);border-color: rgba(var(--ice-base-200) / var(--tw-border-opacity));text-align: center;transition-property: color, background-color, border-color, text-decoration-color, fill, stroke, opacity, box-shadow, transform, filter, backdrop-filter;transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);transition-timing-function: cubic-bezier(0, 0, 0.2, 1);transition-duration: 200ms;height: 3rem;min-height: 3rem;padding-left: 1rem;padding-right: 1rem;font-size: 0.875rem;line-height: 1em;--tw-border-opacity: 1;--tw-bg-opacity: 1;background-color: rgba(var(--ice-base-200) / var(--tw-bg-opacity));--tw-text-opacity: 1;color: rgba(var(--ice-base-content) / var(--tw-text-opacity));text-decoration-line: none;outline-color: rgba(var(--ice-base-200) / 1);border-width: var(--ice-border-btn, 1px)
}.btn-disabled,.btn[disabled],.btn:disabled {pointer-events: none
}.btn:is(input[type=checkbox]),.btn:is(input[type=radio]) {width: auto;appearance: none
}.btn:is(input[type=checkbox]):after,.btn:is(input[type=radio]):after {--tw-content: attr(aria-label);content: var(--tw-content)
}.btn:focus-visible {outline-style: solid;outline-width: 2px;outline-offset: 2px
}.btn:active {--tw-border-opacity: 1;border-color: rgba(var(--ice-base-300) / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color: rgba(var(--ice-base-300) / var(--tw-bg-opacity))
}.btn.btn-active {--tw-border-opacity: 1;border-color: rgba(var(--ice-base-300) / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color: rgba(var(--ice-base-300) / var(--tw-bg-opacity))
}.btn.glass {--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);outline-color: currentColor
}.btn.glass:active {--ice-glass-opacity: 25%;--ice-glass-border-opacity: 15%
}.btn.glass.btn-active {--ice-glass-opacity: 25%;--ice-glass-border-opacity: 15%
}.btn.btn-disabled,.btn[disabled],.btn:disabled {--tw-border-opacity: 0;background-color: rgba(var(--ice-neutral) / var(--tw-bg-opacity));--tw-bg-opacity: 0.2;color: rgba(var(--ice-base-content) / var(--tw-text-opacity));--tw-text-opacity: 0.2
}.btn:is(input[type=checkbox]:checked),.btn:is(input[type=radio]:checked) {--tw-border-opacity: 1;border-color: rgba(var(--ice-primary) / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color: rgba(var(--ice-primary) / var(--tw-bg-opacity));--tw-text-opacity: 1;color: rgba(var(--ice-primary-content) / var(--tw-text-opacity))
}.btn:is(input[type=checkbox]:checked):active,.btn:is(input[type=radio]:checked):active {--tw-border-opacity: 1;border-color: rgba(var(--ice-primary-focus) / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color: rgba(var(--ice-primary-focus) / var(--tw-bg-opacity))
}.btn:is(input[type=checkbox]:checked):focus-visible,.btn:is(input[type=radio]:checked):focus-visible {outline-color: rgba(var(--ice-primary) / 1)
}@keyframes checkmark {0% {background-position-y: 5px
    }50% {background-position-y: -2px
    }100% {background-position-y: 0
    }
}@keyframes progress-loading {50% {background-position-x: -115%
    }
}@keyframes radiomark {0% {box-shadow: 0 0 0 12px rgba(var(--ice-base-100)) inset, 0 0 0 12px rgba(var(--ice-base-100)) inset
    }50% {box-shadow: 0 0 0 3px rgba(var(--ice-base-100)) inset, 0 0 0 3px rgba(var(--ice-base-100)) inset
    }100% {box-shadow: 0 0 0 4px rgba(var(--ice-base-100)) inset, 0 0 0 4px rgba(var(--ice-base-100)) inset
    }
}@keyframes rating-pop {0% {transform: translateY(-0.125em)
    }40% {transform: translateY(-0.125em)
    }100% {transform: translateY(0)
    }
}@keyframes toast-pop {0% {transform: scale(0.9);opacity: 0
    }100% {transform: scale(1);opacity: 1
    }
}"
`;

exports[`tailwindcss > use plugin case 1 1`] = `
"*, ::before, ::after {--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x:  ;--tw-pan-y:  ;--tw-pinch-zoom:  ;--tw-scroll-snap-strictness: proximity;--tw-gradient-from-position:  ;--tw-gradient-via-position:  ;--tw-gradient-to-position:  ;--tw-ordinal:  ;--tw-slashed-zero:  ;--tw-numeric-figure:  ;--tw-numeric-spacing:  ;--tw-numeric-fraction:  ;--tw-ring-inset:  ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / 0.5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur:  ;--tw-brightness:  ;--tw-contrast:  ;--tw-grayscale:  ;--tw-hue-rotate:  ;--tw-invert:  ;--tw-saturate:  ;--tw-sepia:  ;--tw-drop-shadow:  ;--tw-backdrop-blur:  ;--tw-backdrop-brightness:  ;--tw-backdrop-contrast:  ;--tw-backdrop-grayscale:  ;--tw-backdrop-hue-rotate:  ;--tw-backdrop-invert:  ;--tw-backdrop-opacity:  ;--tw-backdrop-saturate:  ;--tw-backdrop-sepia:  
}::backdrop {--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x:  ;--tw-pan-y:  ;--tw-pinch-zoom:  ;--tw-scroll-snap-strictness: proximity;--tw-gradient-from-position:  ;--tw-gradient-via-position:  ;--tw-gradient-to-position:  ;--tw-ordinal:  ;--tw-slashed-zero:  ;--tw-numeric-figure:  ;--tw-numeric-spacing:  ;--tw-numeric-fraction:  ;--tw-ring-inset:  ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / 0.5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur:  ;--tw-brightness:  ;--tw-contrast:  ;--tw-grayscale:  ;--tw-hue-rotate:  ;--tw-invert:  ;--tw-saturate:  ;--tw-sepia:  ;--tw-drop-shadow:  ;--tw-backdrop-blur:  ;--tw-backdrop-brightness:  ;--tw-backdrop-contrast:  ;--tw-backdrop-grayscale:  ;--tw-backdrop-hue-rotate:  ;--tw-backdrop-invert:  ;--tw-backdrop-opacity:  ;--tw-backdrop-saturate:  ;--tw-backdrop-sepia:  
}.btn {display: inline-flex;flex-shrink: 0;cursor: pointer;user-select: none;flex-wrap: wrap;align-items: center;justify-content: center;border-radius: var(--ice-rounded-btn, 0.5rem);border-color: rgba(var(--ice-base-200) / var(--tw-border-opacity));text-align: center;transition-property: color, background-color, border-color, text-decoration-color, fill, stroke, opacity, box-shadow, transform, filter, backdrop-filter;transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);transition-timing-function: cubic-bezier(0, 0, 0.2, 1);transition-duration: 200ms;height: 3rem;min-height: 3rem;padding-left: 1rem;padding-right: 1rem;font-size: 0.875rem;line-height: 1em;--tw-border-opacity: 1;--tw-bg-opacity: 1;background-color: rgba(var(--ice-base-200) / var(--tw-bg-opacity));--tw-text-opacity: 1;color: rgba(var(--ice-base-content) / var(--tw-text-opacity));text-decoration-line: none;outline-color: rgba(var(--ice-base-200) / 1);border-width: var(--ice-border-btn, 1px)
}.btn-disabled,.btn[disabled],.btn:disabled {pointer-events: none
}.btn:is(input[type=checkbox]),.btn:is(input[type=radio]) {width: auto;appearance: none
}.btn:is(input[type=checkbox]):after,.btn:is(input[type=radio]):after {--tw-content: attr(aria-label);content: var(--tw-content)
}.btn:focus-visible {outline-style: solid;outline-width: 2px;outline-offset: 2px
}.btn:active {--tw-border-opacity: 1;border-color: rgba(var(--ice-base-300) / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color: rgba(var(--ice-base-300) / var(--tw-bg-opacity))
}.btn.btn-active {--tw-border-opacity: 1;border-color: rgba(var(--ice-base-300) / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color: rgba(var(--ice-base-300) / var(--tw-bg-opacity))
}.btn.glass {--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);outline-color: currentColor
}.btn.glass:active {--ice-glass-opacity: 25%;--ice-glass-border-opacity: 15%
}.btn.glass.btn-active {--ice-glass-opacity: 25%;--ice-glass-border-opacity: 15%
}.btn.btn-disabled,.btn[disabled],.btn:disabled {--tw-border-opacity: 0;background-color: rgba(var(--ice-neutral) / var(--tw-bg-opacity));--tw-bg-opacity: 0.2;color: rgba(var(--ice-base-content) / var(--tw-text-opacity));--tw-text-opacity: 0.2
}.btn:is(input[type=checkbox]:checked),.btn:is(input[type=radio]:checked) {--tw-border-opacity: 1;border-color: rgba(var(--ice-primary) / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color: rgba(var(--ice-primary) / var(--tw-bg-opacity));--tw-text-opacity: 1;color: rgba(var(--ice-primary-content) / var(--tw-text-opacity))
}.btn:is(input[type=checkbox]:checked):active,.btn:is(input[type=radio]:checked):active {--tw-border-opacity: 1;border-color: rgba(var(--ice-primary-focus) / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color: rgba(var(--ice-primary-focus) / var(--tw-bg-opacity))
}.btn:is(input[type=checkbox]:checked):focus-visible,.btn:is(input[type=radio]:checked):focus-visible {outline-color: rgba(var(--ice-primary) / 1)
}@keyframes checkmark {0% {background-position-y: 5px
    }50% {background-position-y: -2px
    }100% {background-position-y: 0
    }
}@keyframes progress-loading {50% {background-position-x: -115%
    }
}@keyframes radiomark {0% {box-shadow: 0 0 0 12px rgba(var(--ice-base-100)) inset, 0 0 0 12px rgba(var(--ice-base-100)) inset
    }50% {box-shadow: 0 0 0 3px rgba(var(--ice-base-100)) inset, 0 0 0 3px rgba(var(--ice-base-100)) inset
    }100% {box-shadow: 0 0 0 4px rgba(var(--ice-base-100)) inset, 0 0 0 4px rgba(var(--ice-base-100)) inset
    }
}@keyframes rating-pop {0% {transform: translateY(-0.125em)
    }40% {transform: translateY(-0.125em)
    }100% {transform: translateY(0)
    }
}@keyframes toast-pop {0% {transform: scale(0.9);opacity: 0
    }100% {transform: scale(1);opacity: 1
    }
}.btn::after {border: none
}"
`;

exports[`tailwindcss > use plugin case 2 1`] = `
"*, ::before, ::after {--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x:  ;--tw-pan-y:  ;--tw-pinch-zoom:  ;--tw-scroll-snap-strictness: proximity;--tw-gradient-from-position:  ;--tw-gradient-via-position:  ;--tw-gradient-to-position:  ;--tw-ordinal:  ;--tw-slashed-zero:  ;--tw-numeric-figure:  ;--tw-numeric-spacing:  ;--tw-numeric-fraction:  ;--tw-ring-inset:  ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / 0.5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur:  ;--tw-brightness:  ;--tw-contrast:  ;--tw-grayscale:  ;--tw-hue-rotate:  ;--tw-invert:  ;--tw-saturate:  ;--tw-sepia:  ;--tw-drop-shadow:  ;--tw-backdrop-blur:  ;--tw-backdrop-brightness:  ;--tw-backdrop-contrast:  ;--tw-backdrop-grayscale:  ;--tw-backdrop-hue-rotate:  ;--tw-backdrop-invert:  ;--tw-backdrop-opacity:  ;--tw-backdrop-saturate:  ;--tw-backdrop-sepia:  
}::backdrop {--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x:  ;--tw-pan-y:  ;--tw-pinch-zoom:  ;--tw-scroll-snap-strictness: proximity;--tw-gradient-from-position:  ;--tw-gradient-via-position:  ;--tw-gradient-to-position:  ;--tw-ordinal:  ;--tw-slashed-zero:  ;--tw-numeric-figure:  ;--tw-numeric-spacing:  ;--tw-numeric-fraction:  ;--tw-ring-inset:  ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / 0.5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur:  ;--tw-brightness:  ;--tw-contrast:  ;--tw-grayscale:  ;--tw-hue-rotate:  ;--tw-invert:  ;--tw-saturate:  ;--tw-sepia:  ;--tw-drop-shadow:  ;--tw-backdrop-blur:  ;--tw-backdrop-brightness:  ;--tw-backdrop-contrast:  ;--tw-backdrop-grayscale:  ;--tw-backdrop-hue-rotate:  ;--tw-backdrop-invert:  ;--tw-backdrop-opacity:  ;--tw-backdrop-saturate:  ;--tw-backdrop-sepia:  
}.btn {display: inline-flex;flex-shrink: 0;cursor: pointer;user-select: none;flex-wrap: wrap;align-items: center;justify-content: center;border-radius: var(--ice-rounded-btn, 0.5rem);border-color: rgba(var(--ice-base-200) / var(--tw-border-opacity));text-align: center;transition-property: color, background-color, border-color, text-decoration-color, fill, stroke, opacity, box-shadow, transform, filter, backdrop-filter;transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);transition-timing-function: cubic-bezier(0, 0, 0.2, 1);transition-duration: 200ms;height: 3rem;min-height: 3rem;padding-left: 1rem;padding-right: 1rem;font-size: 0.875rem;line-height: 1em;--tw-border-opacity: 1;--tw-bg-opacity: 1;background-color: rgba(var(--ice-base-200) / var(--tw-bg-opacity));--tw-text-opacity: 1;color: rgba(var(--ice-base-content) / var(--tw-text-opacity));text-decoration-line: none;outline-color: rgba(var(--ice-base-200) / 1);border-width: var(--ice-border-btn, 1px)
}.btn-disabled,.btn[disabled],.btn:disabled {pointer-events: none
}.btn:is(input[type=checkbox]),.btn:is(input[type=radio]) {width: auto;appearance: none
}.btn:is(input[type=checkbox]):after,.btn:is(input[type=radio]):after {--tw-content: attr(aria-label);content: var(--tw-content)
}.btn:focus-visible {outline-style: solid;outline-width: 2px;outline-offset: 2px
}.btn:active {--tw-border-opacity: 1;border-color: rgba(var(--ice-base-300) / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color: rgba(var(--ice-base-300) / var(--tw-bg-opacity))
}.btn.btn-active {--tw-border-opacity: 1;border-color: rgba(var(--ice-base-300) / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color: rgba(var(--ice-base-300) / var(--tw-bg-opacity))
}.btn.glass {--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);outline-color: currentColor
}.btn.glass:active {--ice-glass-opacity: 25%;--ice-glass-border-opacity: 15%
}.btn.glass.btn-active {--ice-glass-opacity: 25%;--ice-glass-border-opacity: 15%
}.btn.btn-disabled,.btn[disabled],.btn:disabled {--tw-border-opacity: 0;background-color: rgba(var(--ice-neutral) / var(--tw-bg-opacity));--tw-bg-opacity: 0.2;color: rgba(var(--ice-base-content) / var(--tw-text-opacity));--tw-text-opacity: 0.2
}.btn:is(input[type=checkbox]:checked),.btn:is(input[type=radio]:checked) {--tw-border-opacity: 1;border-color: rgba(var(--ice-primary) / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color: rgba(var(--ice-primary) / var(--tw-bg-opacity));--tw-text-opacity: 1;color: rgba(var(--ice-primary-content) / var(--tw-text-opacity))
}.btn:is(input[type=checkbox]:checked):active,.btn:is(input[type=radio]:checked):active {--tw-border-opacity: 1;border-color: rgba(var(--ice-primary-focus) / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color: rgba(var(--ice-primary-focus) / var(--tw-bg-opacity))
}.btn:is(input[type=checkbox]:checked):focus-visible,.btn:is(input[type=radio]:checked):focus-visible {outline-color: rgba(var(--ice-primary) / 1)
}@keyframes checkmark {0% {background-position-y: 5px
    }50% {background-position-y: -2px
    }100% {background-position-y: 0
    }
}@keyframes progress-loading {50% {background-position-x: -115%
    }
}@keyframes radiomark {0% {box-shadow: 0 0 0 12px rgba(var(--ice-base-100)) inset, 0 0 0 12px rgba(var(--ice-base-100)) inset
    }50% {box-shadow: 0 0 0 3px rgba(var(--ice-base-100)) inset, 0 0 0 3px rgba(var(--ice-base-100)) inset
    }100% {box-shadow: 0 0 0 4px rgba(var(--ice-base-100)) inset, 0 0 0 4px rgba(var(--ice-base-100)) inset
    }
}@keyframes rating-pop {0% {transform: translateY(-0.125em)
    }40% {transform: translateY(-0.125em)
    }100% {transform: translateY(0)
    }
}@keyframes toast-pop {0% {transform: scale(0.9);opacity: 0
    }100% {transform: scale(1);opacity: 1
    }
}.btn::after {border: none
}"
`;
